### DISCUSSION TITLE
Extracting features from built-in networks

### DISCUSSION THREAD

**Discussion by amortazi** [Project-MONAI/tutorials#504] [Q&A]
Hello, 
I am wondering if that's possible to return the outputs of specific layer (for example one of fully connected layers in DenseNet) instead of output of classifier layer in built-in architectures in MONAI like Densenet121? 

Thanks,
Ali

**Comment by ericspod** [comment]
We don't provide that mechanism directly but I think you'd want to use [forward hooks](https://pytorch.org/tutorials/beginner/former_torchies/nnft_tutorial.html#forward-and-backward-function-hooks) to do this.

**Reply by rijobro** [reply]
Is this functionality not supported as part of GradCAM? We have the `ModelWithHooks` class there.
